Source/commands/Command.ts

```diff
@@ -1,33 +1,40 @@
 /*---------------------------------------------------------------------------------------------
- *  Copyright (c) Dolittle. All rights reserved.
- *  Licensed under the MIT License. See LICENSE in the project root for license information.
- *--------------------------------------------------------------------------------------------*/
+*  Copyright (c) Dolittle. All rights reserved.
+*  Licensed under the MIT License. See LICENSE in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { Guid } from '@dolittle/core';
+import { ICommand } from "./internal";
+
 /**
- * Defines the base of a command
+ * Represents a base implementation of {ICommand}
+ *
+ * @export
+ * @abstract
+ * @class Command
+ * @implements {ICommand}
  */
- export class Command {
-    type: string='';
-    defaultValues : any ={};
+export abstract class Command implements ICommand {
```

Comment:
Since we don’t have anything we require anyone deriving from this to implement anything, I suggest we drop the keyword abstract.

Reply:
Hmm, I don’t think that it makes sense. Since I think that under no circumstance will just a “Command” be handled. It enforces good habits by enforcing declarative artifacts in your code base.

Reply:
It is not an abstract class and the keyword abstract implies that there is something the deriving class should implement.

Reply:
Just to clarify; I’m talking about removing the keyword abstract

Reply:
I think we have different perspectives on this.
1) I think of abstract classes of classes that should not be materialized on their own
2) So unless a class does not have behaviour (abstract methods) it shouldn’t be abstract? So data structures (which this is) should not be abstract?

Reply:
I’m not religious on this, I’m just curious since I think this is perfectly fine and makes sense in the context og the consumer and our platform

Reply:
Doesn't this class work more as a base class than an abstract class?

Reply:
What’s the difference